tdrs-frontend/src/components/Reports/Reports.jsx

```diff
@@ -280,18 +300,55 @@ function Reports() {
               </select>
             </label>
           </div>
-          <Button className="margin-y-4" type="button" onClick={handleSearch}>
+          <Button
+            className="margin-y-4"
+            type="button"
+            onClick={() => {
+              if (uploadedFiles.length > 0) {
+                setErrorModalVisible(true)
+              } else {
+                handleSearch()
+              }
+            }}
+          >
             Search
           </Button>
         </form>
       </div>
       {isUploadReportToggled && (
         <UploadReport
           stt={stt?.id}
-          header={submittedHeader}
-          handleCancel={() => setIsToggled(false)}
+          header={`${currentStt} - Fiscal Year ${selectedYear} - ${quarters[selectedQuarter]}`}
+          handleCancel={() => {
+            setIsToggled(false)
+            resetPreviousValues()
+            dispatch(clearFileList())
```

Comment:
make sure files are cleared on cancel so the modal doesn't display if the user tries to execute a new search